osu.Game/Graphics/UserInterface/OsuButton.cs

```diff
@@ -175,6 +175,17 @@ protected override void OnMouseUp(MouseUpEvent e)
             base.OnMouseUp(e);
         }
 
+        public new Axes AutoSizeAxes
+        {
+            get => base.AutoSizeAxes;
+            set
+            {
+                base.AutoSizeAxes = value;
+                Content.RelativeSizeAxes = ~value;
+                Content.AutoSizeAxes = value;
+            }
+        }
+
```

Comment:
Bit hesitant to add something this like. You can probably just specify the size rather than relying on `AutoSize` here?

Reply:
I will probably face the issue that too long string overflow it while too short looks ugly. The same is as with input reset mini-buttons in settings. Maybe, just move this to local buttons (private submit/cancel classes) instead?

Reply:
With that argument, it's fair to say that a lot of buttons are broken in a similar way. I don't think fixing this locally makes sense – we'd want to address this across all buttons in a standardised way.

So I'd recommend just choosing a fixed width for this PR and keeping this until later.